server/text_generation_server/models/__init__.py

```diff
@@ -412,9 +412,8 @@ def get_model(
     sliding_window = config_dict.get("sliding_window", -1)
     if sliding_window != -1 and not SUPPORTS_WINDOWING:
         logger.warning(
-            f"Flash attention is available, but doesn't support windowing which is required by model {model_id}"
+            f"Flash attention is available, but doesn't support windowing which is required by model {model_id} for long contexts."
         )
-        FLASH_ATTENTION = False
```

Comment:
This modifies the behavior of all targets, not just rocm, change things in Rocm if you wnat to change behavior there.

IMHO having load time crash is much better than random runitime one (and checking against max-total-tokens would definitely make sense).


We could change here because WINDOWING is just not needed if `max-total-tokens` < `sliding_window`. (and therefore that logic would be valid for all targets)

Reply:
that works